dag-syntax.cddl

```diff
@@ -0,0 +1,24 @@
+application = {
+    applicationName: tstr,
+    nodes: [ * node ],
+}
+
+node = {
+    task: tstr,
+    name: tstr,
+    type: "producer" / "prosumer" / "consumer",
```

Comment:
Isn't the `type` field a bit redundant? The type seems to be implicit based on the task's (and its potential upstreams') `downstream` value.

Keeping this field in adds potential issues if the task's `type` doesn't match the `downstream` attribute.

Reply:
True, we could remove that.

```suggestion
```